fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/ContainerServiceImpl.java

```diff
@@ -63,7 +63,12 @@ public Container findOrCreate(final Session session, final String path) {
                 initializeNewObjectProperties(node);
             }
 
-            return new ContainerImpl(node);
+            final ContainerImpl container = new ContainerImpl(node);
```

Comment:
It's not quite as easy as that, because `AbstractService.findOrCreatNode()` works with Nodes instead of FedoraResources.  But it's easy enough to call touch() from both the Container and Binary services.